### PR TITLE
Use vector to avoid leaking streaming EPT buffers in case of exception.

### PR DESCRIPTION
**Pull request by connormanning**
Related to #2603.

### CODE REVIEW

File: io/EptReader.cpp [hunk 1]
```diff
@@ -818,11 +818,10 @@ void EptReader::loadNextOverlap()
 void EptReader::fillPoint(PointRef& point)
 {
     DimTypeList dims = m_bufferPointView->dimTypes();
-    char* buffer = new char[m_bufferPointView->pointSize()];
-    m_bufferPointView->getPackedPoint(dims, m_currentIndex, buffer);
-    point.setPackedData(dims, buffer);
+    std::vector<char> buffer(m_bufferPointView->pointSize());
```
Comment (abellgithub) [inline io/EptReader.cpp]: This is fine, but it still seems like it should be done once -- the point size is fixed.

Comment (pravinshinde825) [reply io/EptReader.cpp]: @connormanning , We can make it as a member of EptReader class like: `std::vector<char> m_temp_buffer` and reserve memory for it in `if(pipelineStreamable)` at line 441 in `addDimensions()` like : `m_temp_buffer.reserve(layout->pointSize())`. So here we only need to use `m_temp_buffer.data()`.
@abellgithub , In this way buffer will be allocated only once.

Comment (pravinshinde825) [reply io/EptReader.cpp]: @connormanning , Shall I do that??

Comment (connormanning) [reply io/EptReader.cpp]: Go ahead if you'd like @pravinshinde825, my only concern for this PR was the `new`/`delete`.

Comment (pravinshinde825) [reply io/EptReader.cpp]: @connormanning , Please check https://github.com/PDAL/PDAL/pull/2617

**Pull request merged**